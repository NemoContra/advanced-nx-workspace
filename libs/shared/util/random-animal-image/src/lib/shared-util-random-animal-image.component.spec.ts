import {
  AnimalImage,
  SharedUtilRandomAnimalImageComponent,
  SharedUtilRandomAnimalImageService,
} from '@flight-workspace/shared/util/random-animal-image';
import { createComponentFactory, mockProvider } from '@ngneat/spectator/jest';
import { of } from 'rxjs';

describe('SharedUtilRandomAnimalImageComponent', () => {
  const createComponent = createComponentFactory({
    component: SharedUtilRandomAnimalImageComponent,
    providers: [
      mockProvider(SharedUtilRandomAnimalImageService, {
        getAnimalImage: jest
          .fn()
          .mockReturnValue(
            of({
              src: 'http://someimage.com/someimage.png',
              alt: 'TEST',
            } as AnimalImage)
          ),
      }),
    ],
  });

  it('should render an image with src and alt', () => {
    const spectator = createComponent();
    expect(spectator.fixture).toMatchSnapshot();
  });
});
