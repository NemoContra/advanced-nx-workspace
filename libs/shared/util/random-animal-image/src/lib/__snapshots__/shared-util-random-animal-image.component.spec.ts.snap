// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SharedUtilRandomAnimalImageComponent should render an image with src and alt 1`] = `
<flight-workspace-shared-util-random-animal-image
  __ngContext__={[Function LRootView]}
  img$={[Function Observable]}
  sharedUtilRandomAnimalImageService={[Function Object]}
>
  <img
    alt="TEST"
    height="500px"
    src="http://someimage.com/someimage.png"
    width="500px"
  />
</flight-workspace-shared-util-random-animal-image>
`;
